libretro-common/string/stdstring.c

```diff
@@ -534,3 +535,183 @@ unsigned string_hex_to_unsigned(const char *str)
 }
+
+/**
+ * Get the total number of occurrences of a character in the given string.
+ */
+int count_occurrences_single_character(char *str, char t)
+{
+   int ctr = 0;
+   int i;
+
+   for (i = 0; str[i] != '\0'; ++i) {
+      if (t == str[i])
+         ++ctr;
+   }
+
+   return ctr;
+}
+
+/**
+ * Replaces all spaces with the given character.
+ */
+void replace_whitespace_with_single_character(char *str, char t)
+{
+
+   while (*str) {
+      if (isspace(*str))
+         *str = t;
+      str++;
+   }
+}
+
+/**
+ * Replaces multiple spaces with a single space in a string.
+ */
+void replace_multi_space_with_single_space(char *str)
+{
+   char *dest = str;
+
+   while (*str != '\0')
+   {
+      while (*str == ' ' && *(str + 1) == ' ')
+         str++;
+
+      *dest++ = *str++;
+   }
+
+   *dest = '\0';
+}
+
+/**
+ * Remove all spaces from the given string.
+ */
+void remove_all_whitespace (char* str_trimmed, const char* str_untrimmed)
+{
+   while (*str_untrimmed != '\0')
+   {
+      if(!isspace(*str_untrimmed))
+      {
+         *str_trimmed = *str_untrimmed;
+         str_trimmed++;
+      }
+      str_untrimmed++;
+   }
+   *str_trimmed = '\0';
+}
+
+/**
+ * Retrieve the last occurance of the given character in a string.
+ */
+int index_last_occurance(char str[], char t)
+{
+   const char * ret = strrchr(str, t);
+   if (ret)
+      return ret-str;
+
+   return -1;
+}
+
+/**
+ * Find the position of a substring in a string.
+ */
+int find_index_substring_string(const char* str1, const char* str2)
+{
+   int index;
+
+   if (str1[0] != '\0')
+   {
+      const char *pfound = strstr(str1, str2);
+      if (pfound != NULL)
+      {
+         index = (pfound - str1);
+         return index;
+      }
+   }
+
+   return -1;
+}
+
+/**
+ * Given a filename and position, find the associated disc number.
+ */
+int find_disc_number(const char* str1, int index)
+{
+   char disc;
+   int disc_number = 0;
+
+   disc = str1[index + 6];
+
+   switch(disc)
+   {
+      case 'a':
+      case 'A':
+         disc_number = 1;
+         break;
+      case 'b':
+      case 'B':
+         disc_number = 2;
+         break;
+      case 'c':
+      case 'C':
+         disc_number = 3;
+         break;
+      case 'd':
+      case 'D':
+         disc_number = 4;
+         break;
+      case 'e':
+      case 'E':
+         disc_number = 5;
+         break;
+      case 'f':
+      case 'F':
+         disc_number = 6;
+         break;
+      case 'g':
+      case 'G':
+         disc_number = 7;
+         break;
+      case 'h':
+      case 'H':
+         disc_number = 8;
+         break;
+      case 'i':
+      case 'I':
+         disc_number = 9;
+         break;
+      default:
+         disc_number = disc - '0';
+         break;
+   }
+
+   if (disc_number >= 1)
+      return disc_number;
+
+   return 0;
+}
+
+/**
+ * Given a title and filename, append the appropriate disc number to it.
+ */
+void append_multi_disc_suffix(char * str1, const char *filename)
```

Comment:
`append_multi_disc_suffix` seems outside the scope of the generic stdstring functions. May want to have it be in the database scanner c file too.